### PR TITLE
Add type hints for pytorch models (final batch)

### PR DESCRIPTION
**Pull request by nablabits**
# What does this PR do?
Addresses the last type hints for pytorch models in https://github.com/huggingface/transformers/issues/16059:

1. TableTransformer: 
	1. `TableTransformerForObjectDetection`
	2. `TableTransformerModel`
2. `TimesformerModel`
3. `TimmBackbone`
4. TVLT: 
	1. `TvltForAudioVisualClassification`
	2. `TvltForPreTraining`
	3. `TvltModel`
5. Vivit
	1. `VivitForVideoClassification`
	2. `VivitModel`

## Who can review?
@Rocketknight1 please

### CODE REVIEW

File: src/transformers/models/table_transformer/modeling_table_transformer.py [hunk 1]
```diff
@@ -814,15 +814,15 @@ def _set_gradient_checkpointing(self, module, value=False):
 
             Pixel values can be obtained using [`DetrImageProcessor`]. See [`DetrImageProcessor.__call__`] for details.
 
-        pixel_mask (`torch.LongTensor` of shape `(batch_size, height, width)`, *optional*):
+        pixel_mask (`torch.FloatTensor` of shape `(batch_size, height, width)`, *optional*):
```
Comment (nablabits) [inline src/transformers/models/table_transformer/modeling_table_transformer.py:817]: Here we are facing the same issue as in [here](https://github.com/huggingface/transformers/pull/25557#discussion_r1300015531) I've changed the docstring as the model seems to work with both types. In the case of the `decoder_attention_mask`, that is not used by the method so one can pass whatever she wants.

Comment (Rocketknight1) [reply src/transformers/models/table_transformer/modeling_table_transformer.py:817]: Yep, I think `FloatTensor` is correct here!

File: src/transformers/models/vivit/modeling_vivit.py [hunk 2]
```diff
@@ -489,12 +489,12 @@ def _prune_heads(self, heads_to_prune):
     @replace_return_docstrings(output_type=BaseModelOutput, config_class=_CONFIG_FOR_DOC)
```
Comment (Rocketknight1) [inline src/transformers/models/vivit/modeling_vivit.py:489]:
```suggestion
    @replace_return_docstrings(output_type=BaseModelOutputWithPooling, config_class=_CONFIG_FOR_DOC)
```

Comment (Rocketknight1) [reply src/transformers/models/vivit/modeling_vivit.py:489]: This isn't your fault again, but I notice the docstring decorator has the wrong type here - your type hints are correct!

**Review by Rocketknight1 — APPROVED** [review]
LGTM!

**Comment by Rocketknight1** [comment]
Thank you again for the diligent work on the type hints, by the way - your PRs were really clean and required very little work to review, and they really helped push that project!

File: src/transformers/models/vivit/modeling_vivit.py (hunk 2 above)
Comment (nablabits) [reply src/transformers/models/vivit/modeling_vivit.py:489]: Interesting, I'll try to catch these guys in the future

**Comment by nablabits** [comment]
> Thank you again for the diligent work on the type hints, by the way - your PRs were really clean and required very little work to review, and they really helped push that project!

Thank you very much, really appreciated!! TBH it has been a great learning experience for me, so thanks for the opportunity and the patience with the explanations

**Comment by SoyGema** [comment]
> Thank you again for the diligent work on the type hints, by the way - your PRs were really clean and required very little work to review, and they really helped push that project!

My apologies, in advance, for stepping in here. 
I would like to highlight  as well  the positive impact  @nablabits  brings to the HF OSS ecosystem. 
It´s also great how this user helps other contributors to thrive  [#1](https://github.com/SoyGema/contrib_schema/issues/1) and [#2](https://github.com/SoyGema/contrib_schema/issues/2) bringing learning, reciprocity, and fun.

**Comment by nablabits** [comment]
> > Thank you again for the diligent work on the type hints, by the way - your PRs were really clean and required very little work to review, and they really helped push that project!
> 
> My apologies, in advance, for stepping in here. I would like to highlight as well the positive impact @nablabits brings to the HF OSS ecosystem. It´s also great how this user helps other contributors to thrive [#1](https://github.com/SoyGema/contrib_schema/issues/1) and [#2](https://github.com/SoyGema/contrib_schema/issues/2) bringing learning, reciprocity, and fun.

Gema, thanks for the kind words :hugs: 
@SoyGema

**Pull request merged**